spec/invoker/power/url_rewriter_spec.rb

```diff
@@ -66,5 +79,98 @@
         Invoker.config = @original_invoker_config
       end
     end
+
+    context '#select_backend_config' do
+      before(:all) do
+        @original_invoker_config = Invoker.dns_cache
+        @processes = [
+          { label: 'foo',     port: 1 },
+          { label: 'api.foo', port: 2 },
+          { label: 'bar.foo', port: 3 },
+        ].map{ |p| OpenStruct.new(p) }
+        Invoker.config.stubs(:processes).returns(@processes)
+        Invoker.dns_cache = Invoker::DNSCache.new(nil)
+      end
+
+      before do
+        def rewriter.dns_check(*args)
+          socket = StringIO.new
+          Invoker::IPC::DnsCheckCommand.new(socket).run_command(Invoker::IPC::Message::DnsCheck.new(*args))
+          socket.rewind
+          Invoker::IPC.message_from_io socket
+        end
+      end
+
+      after(:all) do
+        Invoker.dns_cache = @original_invoker_config
+      end
+
+      it 'matches foo.dev' do
+        match = rewriter.select_backend_config('foo.dev')
+        expect(match.port).to eql(1)
+      end
+
+      it 'matches api.foo.dev' do
+        match = rewriter.select_backend_config('api.foo.dev')
+        expect(match.port).to eql(2)
+      end
+
+      it 'matches bar.foo.dev' do
+        match = rewriter.select_backend_config('bar.foo.dev')
+        expect(match.port).to eql(3)
+      end
+
+      it 'matches baz.foo.dev' do
+        match = rewriter.select_backend_config('baz.foo.dev')
+        expect(match.port).to eql(1)
+      end
+
+      it 'does not match baz.dev' do
+        match = rewriter.select_backend_config('baz.dev')
+        expect(match.port).to be_nil
+      end
+
+      context 'with an alias' do
+        it 'matches baz.foo.dev to the alias' do
+          @processes[1][:alias] = '*.foo'
+          Invoker.dns_cache = Invoker::DNSCache.new(nil)
+
+          match = rewriter.select_backend_config('baz.foo.dev')
+          expect(match.port).to eql(2)
+        end
+
+        it 'matches baz.dev to the alias' do
+          @processes[1][:alias] = '*'
+          Invoker.dns_cache = Invoker::DNSCache.new(nil)
+
+          match = rewriter.select_backend_config('baz.dev')
+          expect(match.port).to eql(2)
```

Comment:
The original domain is `api.foo` and specifying an alias of `*` takes over a  unrelated domain `baz.dev`. So what happens if more than one entry specifies `*` alias?

Reply:
right now, the first entry with `*` will match and any others will be ignored.